packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.ts

```diff
@@ -87,36 +89,52 @@ export class UiSettingsService
     this.validatesOverrides();
 
     return {
-      asScopedToClient: this.getScopedClientFactory(),
+      asScopedToClient: this.getScopedClientFactory('namespace'),
+      asScopedToGlobalClient: this.getScopedClientFactory('global'),
     };
   }
 
   public async stop() {}
 
-  private getScopedClientFactory(): (
-    savedObjectsClient: SavedObjectsClientContract
-  ) => UiSettingsClient {
+  private getScopedClientFactory(
+    scope: UiSettingsScope
+  ): (savedObjectsClient: SavedObjectsClientContract) => UiSettingsClient | UiSettingsGlobalClient {
     const { version, buildNum } = this.coreContext.env.packageInfo;
-    return (savedObjectsClient: SavedObjectsClientContract) =>
-      new UiSettingsClient({
-        type: 'config',
+    return (savedObjectsClient: SavedObjectsClientContract) => {
+      const isNamespaceScope = scope === 'namespace';
+
+      const options = {
+        type: (isNamespaceScope ? 'config' : 'config-global') as 'config' | 'config-global',
         id: version,
         buildNum,
         savedObjectsClient,
-        defaults: mapToObject(this.uiSettingsDefaults),
-        overrides: this.overrides,
+        defaults: isNamespaceScope
+          ? mapToObject(this.uiSettingsDefaults)
+          : mapToObject(this.uiSettingsGlobalDefaults),
+        overrides: isNamespaceScope ? this.overrides : {},
         log: this.log,
-      });
+      };
+      return UiSettingsClientFactory.create(options);
+    };
   }
 
-  private register(settings: Record<string, UiSettingsParams> = {}) {
+  private register = (settings: Record<string, UiSettingsParams> = {}) => {
     Object.entries(settings).forEach(([key, value]) => {
       if (this.uiSettingsDefaults.has(key)) {
         throw new Error(`uiSettings for the key [${key}] has been already registered`);
       }
       this.uiSettingsDefaults.set(key, value);
     });
-  }
+  };
+
+  private registerGlobal = (settings: Record<string, UiSettingsParams> = {}) => {
```

Comment:
I know we talked about making the key unique across the two settings, but given we have two SO now, I don't see a need for that?

Reply:
I agree, given the stricter separation we went with, I think it's fine.